src/mud/examples/poisson.py

```diff
@@ -126,13 +126,99 @@ def spline_objective_function_2d(lam, aff=10000):
     return np.linalg.norm(g - vals)
 
 
+def plot_qoi_ex(mud_prob, num_components: int, **kwargs):
+    fig = plt.figure(figsize=(10, 5))
+    for i in range(num_components):
+        ax = fig.add_subplot(1, 2, i + 1)
+        mud_prob.plot_params_2d(ax=ax, y=i, contours=True, colorbar=True)
+        if i == 1:
+            ax.set_ylabel("")
+    if kwargs.get("save_path"):
+        save_figure("learned_qoi", **kwargs)
+    return ax
+
+
+def plot_response_ex(
+    poisson_prob,
+    closest,
+    raw_data,
+    sensors_mask,
+    group_idxs: List[List[int]] = [[0, 5], [5, 50], [50, -1]],
+    markers: List[str] = ["*", "+", "."],
+    colors: List[str] = ["red", "white", "k"],
+    **kwargs,
+):
+    fig = plt.figure(figsize=(10, 5))
+    ax = fig.add_subplot(1, 2, 1)
+
+    # Plot response surface from solving Eq. 30.
+    # u stores the mesh and values as solved by fenics
+    mesh, vals = raw_data["u"]
+    tcf = ax.tricontourf(mesh[:, 0], mesh[:, 1], vals, levels=20, vmin=-0.5, vmax=0)
+    fig.colorbar(tcf)
+
+    # Plot points used for each ordering
+    for idx, oi in enumerate(group_idxs):
+        start_idx, end_idx = group_idxs[idx][0], group_idxs[idx][1]
+        mask = sensors_mask[start_idx:end_idx]
+        poisson_prob.sensor_scatter_plot(
+            ax=ax,
+            mask=mask,
+            color=colors[idx],
+            marker=markers[idx],
+        )
+
+    # Label and format figure
+    _ = plt.xlim(0, 1)
+    _ = plt.ylim(0, 1)
+    _ = plt.title("Response Surface")
+    _ = plt.xlabel("$x_1$")
+    _ = plt.ylabel("$x_2$")
+
+    ax = fig.add_subplot(1, 2, 2)
+    # Plot closest solution in sample set to the reference solution
+    plot_solution_spline(
+        closest,
+        ax=ax,
+        lw=5,
+        c="green",
+        ls="--",
+        label=r"$\hat{g}(\lambda^\dagger)$",
+        zorder=50,
+    )
+    # Plot first 100 lambda initial
+    for i, lam in enumerate(poisson_prob.lam[0:50]):
+        plot_solution_spline(
+            lam,
+            plot_true=False,
+            ax=ax,
+            lw=1,
+            c="purple",
+            alpha=0.1,
+        )
+
+    plot_vert_line(ax, 1 / 3.0, color="b", linestyle="--", alpha=0.6)
+    plot_vert_line(ax, 2 / 3.0, color="b", linestyle="--", alpha=0.6)
+
+    ax.set_title("Boundary Condition")
+    ax.set_ylabel("")
+    ax.legend(
+        ["$g(x_2)$", r"$\hat{g}(x_2,\lambda^\dagger)$", r"$\hat{g}(x_2,\lambda_i)$"]
+    )
+
+    fig.tight_layout()
+    if kwargs.get("save_path"):
```

Comment:
Should be 
`if kwargs,get("save_path") is not None:`
See [PEP 8](https://peps.python.org/pep-0008/)